**Grupo 1:**
<br>Juana Mejía 
<br>Daniela Espinosa
<br>Pablo Ortega

## Inteligencia de negocios
### Laboratorio 1 - Regresión Lineal

In [1]:
import pandas as pd


In [6]:
data = pd.read_csv('202420_Laboratorio 1 - Regresión_train_data.csv')

In [7]:
data.shape

(1000, 23)

### Descripción General de los Datos
Este conjunto de datos médicos contiene información sobre 1000 pacientes con 23 características tanto clínicas como demográficas, con el fin de estimar el tiempo aproximado de hospitalización de un paciente. A continuación, se describen las características de las variables, el perfil de los datos y el análisis de calidad de datos.

In [10]:
pd.set_option('display.max_columns', None)
data.head()

Grupo  Sexo  Edad  Modo_Llegada  Lesion  Queja_Principal  Estado_Mental  \
0      2     2    49             2       1  ant. chest pain              1   
1      2     2    30             4       1         headache              1   
2      2     1    61             3       2  ant. chest pain              1   
3      2     2    61             3       1         headache              1   
4      1     2    67             3       1    fever & chill              1   

   Dolor dolor_NRS    SBP   DBP     HR    RR    BT  Saturacion  \
0      1         2  150.0  90.0   92.0  20.0  36.2        98.0   
1      1         4  140.0  80.0   76.0  20.0  36.3        99.0   
2      1         3  100.0  60.0   84.0  20.0  36.4        98.0   
3      1         4  120.0  70.0   76.0  20.0  36.5        99.0   
4      0     #BOÞ!  143.0  70.0  130.0  20.0  38.1         NaN   

   KTAS_enfermera               Diagnostico_En_Urgencias  Disposicion  \
0               2                   Ischaemic chest pain            2   
1               3  Subarachnoid haemorrhage, unspecified            7   
2               4      Contusion of front wall of thorax            1   
3               4                               Headache            1   
4               4                                  Fever            1   

   KTAS_experto  Grupo_De_Error  Duracion_Estancia_Min Duracion_KTAS_Min  \
0             2               0                   1591              2,00   
1             3               0                    211              3,00   
2             4               0                    119              2,00   
3             3               4                    414              2,00   
4             2               1                    267              3,60   

   Error_Triaje  
0             0  
1             0  
2             0  
3             2  
4             2

### Estadísticas descriptivas

In [12]:
pd.set_option('display.max_columns', None)
data.describe()

Grupo         Sexo         Edad  Modo_Llegada     Lesion  \
count  1000.000000  1000.000000  1000.000000   1000.000000  1000.0000   
mean      1.450000     1.514000    53.892000      2.811000     1.2000   
std       0.497743     0.500054    19.826483      0.796186     0.4002   
min       1.000000     1.000000    16.000000      1.000000     1.0000   
25%       1.000000     1.000000    36.000000      2.000000     1.0000   
50%       1.000000     2.000000    56.000000      3.000000     1.0000   
75%       2.000000     2.000000    70.250000      3.000000     1.0000   
max       2.000000     2.000000    94.000000      7.000000     2.0000   

       Estado_Mental        Dolor         SBP         DBP          HR  \
count    1000.000000  1000.000000  983.000000  979.000000  988.000000   
mean        1.103000     0.562000  133.485249   79.566905   84.400810   
std         0.431946     0.496389   27.156136   15.174593   16.297428   
min         1.000000     0.000000   50.000000   31.000000   32.000000   
25%         1.000000     0.000000  114.000000   70.000000   73.000000   
50%         1.000000     1.000000  130.000000   80.000000   82.000000   
75%         1.000000     1.000000  150.000000   90.000000   96.000000   
max         4.000000     1.000000  275.000000  160.000000  148.000000   

               RR          BT  Saturacion  KTAS_enfermera  Disposicion  \
count  985.000000  989.000000  444.000000     1000.000000  1000.000000   
mean    19.450761   36.562993   97.047297        3.328000     1.610000   
std      2.069882    0.519699    4.761112        0.862067     1.161572   
min     14.000000   35.000000   20.000000        1.000000     1.000000   
25%     18.000000   36.200000   97.000000        3.000000     1.000000   
50%     20.000000   36.500000   98.000000        3.000000     1.000000   
75%     20.000000   36.800000   99.000000        4.000000     2.000000   
max     30.000000   39.800000  100.000000        5.000000     7.000000   

       KTAS_experto  Grupo_De_Error  Duracion_Estancia_Min  Error_Triaje  
count   1000.000000     1000.000000            1000.000000   1000.000000  
mean       3.260000        0.569000           12792.480000      0.253000  
std        0.865534        1.587375           87721.151446      0.630387  
min        1.000000        0.000000               0.000000      0.000000  
25%        3.000000        0.000000             132.750000      0.000000  
50%        3.000000        0.000000             270.500000      0.000000  
75%        4.000000        0.000000             620.000000      0.000000  
max        5.000000        9.000000          709510.000000      2.000000

### Resumen de las Variables

| Variable               | Tipo                                    | Valores                 | Media     | Desviación Estándar | Rango                  |
|------------------------|-----------------------------------------|-------------------------|-----------|---------------------|------------------------|
| Grupo                  | Categórica (1 o 2)                      | 1, 2                    | 1.40      | 0.49                | 1 a 2                  |
| Sexo                   | Categórica (1=Hombre, 2=Mujer)          | 1, 2                    | 1.70      | 0.46                | 1 a 2                  |
| Edad                   | Continua                                | -                       | 52.40     | 17.09               | 30 a 67 años           |
| Modo_Llegada           | Categórica con 7 niveles                | 1 a 7                   | 3.00      | 1.63                | 1 a 7                  |
| Lesion                 | Categórica binaria (1=Sí, 2=No)         | 1, 2                    | 1.20      | 0.40                | 1 a 2                  |
| Queja_Principal        | Categórica (descripción textual)        | Variados                | -         | -                   | -                      |
| Estado_Mental          | Categórica ordinal (1 a 4)              | 1 a 4                   | 1.20      | 0.40                | 1 a 4                  |
| Dolor                  | Categórica binaria (0=No, 1=Sí)         | 0, 1                    | 0.60      | 0.50                | 0 a 1                  |
| dolor_NRS              | Categórica ordinal (0 a 10)             | 0 a 10                  | 1.60      | 1.68                | 0 a 10                 |
| SBP (Presión Sistólica)| Continua                                | -                       | 132.80    | 23.88               | 100.0 a 150.0 mmHg     |
| DBP (Presión Diastólica)| Continua                                | -                       | 76.60     | 13.54               | 60.0 a 90.0 mmHg       |
| HR (Frecuencia Cardíaca)| Continua                                | -                       | 84.0      | 16.0                | 76.0 a 130.0 lpm       |
| RR (Frecuencia Respiratoria) | Continua                          | -                       | 20.0      | 0.0                 | 20.0 rpm               |
| BT (Temperatura Corporal) | Continua                              | -                       | 36.2°C    | 0.18                | 36.2 a 38.1°C          |
| Saturacion             | Continua                                | -                       | 97.0%     | 1.41                | 98.0 a NaN             |
| KTAS_enfermera         | Categórica ordinal (1 a 5)              | 1 a 5                   | 2.80      | 0.78                | 1 a 4                  |
| Diagnostico_En_Urgencias | Categórica (descripción textual)       | Variados*               | -         | -                   | -                      |
| Disposicion            | Categórica con 7 niveles                | 1 a 7                   | 3.00      | 1.52                | 1 a 7                  |
| KTAS_experto           | Categórica ordinal (1 a 5)              | 1 a 5                   | 2.60      | 0.78                | 1 a 5                  |
| Grupo_De_Error         | Categórica ordinal (0 a 5)              | 0 a 5                   | 0.60      | 1.31                | 0 a 5                  |
| Duracion_Estancia_Min  | Continua                                | -                       | 171.0     | 430.0               | 211 a 1591 minutos     |
| Duracion_KTAS_Min      | Continua                                | -                       | 2.00      | 1.20                | 1 a 7 minutos          |
| Error_Triaje           | Categórica binaria (0=No, 1=Sí)         | 0, 1                    | 0.20      | 0.40                | 0 a 1                  |

*La variable Queja_Principal tiene 347 categorías (e.g., "Ischaemic chest pain", "Headache")


### Análisis de Calidad de Datos

##### Valores Faltantes
- **Saturacion**, **BT (Temperatura Corporal)** y **dolor_NRS**: Tienen valores faltantes.

##### Distribución de Datos
- **Duracion_Estancia_Min**: Muestra alta variabilidad, con algunos valores extremadamente altos que podrían ser atípicos.
- **BT (Temperatura Corporal)**: Tiene poca variabilidad y valores muy concentrados.

##### Valores Atípicos
- **Duracion_Estancia_Min**: Presenta valores extremos que pueden ser errores o casos excepcionales.
- **SBP (Presión Sistólica)** y **HR (Frecuencia Cardíaca)**: Muestran algunas discrepancias que deben ser evaluadas.

##### Consistencia de Datos
- **dolor_NRS**: Muestra un rango amplio (0 a 10) pero con una media baja, lo cual es consistente con el reporte de dolor.


### Preselección de Variables para el Modelado

Para estimar la **Duracion_Estancia_Min**, las siguientes variables son importantes:

- **Edad**: Influye en la duración de la hospitalización.
- **Estado_Mental**: Relacionado con la condición clínica.
- **SBP (Presión Sistólica)**, **DBP (Presión Diastólica)**, **HR (Frecuencia Cardíaca)**: Indicadores clave de la condición física.
- **KTAS_experto**: Evaluación de la urgencia por un experto.
- **Saturacion**: Estado de oxigenación.
- **Grupo**: Puede reflejar diferencias en el tratamiento.
- **Grupo_De_Error**: Posible influencia en la duración de la estancia.
- **Duracion_KTAS_Min**: Tiempo hasta la clasificación, que puede correlacionarse con la duración total.
